hail/python/hail/linalg/blockmatrix.py

```diff
@@ -2060,20 +2061,19 @@ def parse_rects(fname):
                 raise ValueError(f'Invalid rectangle file name: {fname}')
             return rect_idx_and_bounds
 
-        rect_files = [file for file in os.listdir(path) if not re.match(r'.*\.crc', file)]
-        rects = [parse_rects(file) for file in rect_files]
+        rect_files = [file['path'] for file in hl.utils.hadoop_ls(path) if not re.match(r'.*\.crc', file['path'])]
+        rects = [parse_rects(os.path.basename(file_path)) for file_path in rect_files]
 
         n_rows = max(rects, key=lambda r: r[2])[2]
         n_cols = max(rects, key=lambda r: r[4])[4]
 
         nd = np.zeros(shape=(n_rows, n_cols))
-        for rect, file in zip(rects, rect_files):
-            file_path = f'{path}/{file}'
-            if binary:
-                rect_data = np.reshape(np.fromfile(file_path), (rect[2]-rect[1], rect[4]-rect[3]))
-            else:
-                rect_data = np.loadtxt(file_path, ndmin=2)
-
+        for rect, file_path in zip(rects, rect_files):
+            with hl.utils.hadoop_open(file_path, 'rb') as f:
```

Comment:
you're probably better off using `hadoop_copy` to localize this stuff - the `hadoop_open` file objects are SUPER slow.

Reply:
interesting, thanks for the tip!